lib/puppet/functions/certs/certificate_subject.rb

```diff
@@ -6,15 +6,25 @@
 #
 Puppet::Functions.create_function(:'certs::certificate_subject') do
   # @param certificate_path
+  # @param include_spaces
+  #   When true, generates subject in RFC2253 format with a space between each element.
+  #   This is equivalent to specifying openssl x509 -in <certificate_path> -nameopt RFC2253,sep_comma_plus_space
   dispatch :certificate_subject do
-    param 'String[1]', :certificate_path
+    required_param 'String[1]', :certificate_path
+    optional_param 'Optional[Boolean]', :include_spaces
     return_type 'Optional[String]'
   end
 
-  def certificate_subject(certificate_path)
+  def certificate_subject(certificate_path, include_spaces = false)
+    unless File.exist?(certificate_path)
+      Puppet.err("The file at #{certificate_path} does not exist, the certificate subject could not be calculated.")
```

Comment:
I added this check and block. I think this is better than a silent race condition that can happen.